src/sync/orderbook.py

```diff
@@ -0,0 +1,125 @@
+"""Main Entry point for app_hash sync"""
+from typing import Any
+
+from dune_client.file.interface import FileIO
+
+from src.fetch.orderbook import OrderbookFetcher
+from src.logger import set_log
+from src.models.batch_rewards_schema import BatchRewards
+from src.models.block_range import BlockRange
+from src.models.order_rewards_schema import OrderRewards
+from src.models.tables import SyncTable
+from src.post.aws import AWSClient
+from src.sync.common import last_sync_block
+from src.sync.config import SyncConfig
+from src.sync.record_handler import RecordHandler
+from src.sync.upload_handler import UploadHandler
+
+log = set_log(__name__)
+
+
+class OrderbookDataHandler(
+    RecordHandler
+):  # pylint:disable=too-few-public-methods,too-many-arguments
+    """
+    This class is responsible for consuming new dune records and missing values from previous runs
+    it attempts to fetch content for them and filters them into "found" and "not found" as necessary
+    """
+
+    def __init__(
+        self,
+        file_manager: FileIO,
+        block_range: BlockRange,
+        sync_table: SyncTable,
+        config: SyncConfig,
+        data_list: list[dict[str, Any]],
+    ):
+        super().__init__(block_range, sync_table, config)
+        log.info(f"Handling {len(data_list)} new records")
+        self.file_manager = file_manager
+        self.data_list = data_list
+
+    def num_records(self) -> int:
+        return len(self.data_list)
+
+    def write_found_content(self) -> None:
+        self.file_manager.write_ndjson(data=self.data_list, name=self.content_filename)
+
+    def write_sync_data(self) -> None:
+        # Only write these if upload was successful.
```

Comment:
Is there a check for this?

Reply:
Not sure I understand the question. Are you asking if this method is tested somewhere?

Reply:
It might be worth mentioning that this file was just renamed from `order_rewards.py` because the content is almost completely reused. Maybe its better if I don't perform the rename here so we do not need to look at logic that has not changed. I do believe this method has been tested, but we may have to double check.

Reply:
From the comment, I understood that this code below is executed only if the upload is successful. I wanted to know if you do check somewhere if the upload was successful because I couldn't find it.

Reply:
Upload being successful is reaching the next point in the linearly executed code. If the upload was not successfull the script run would fail.